### PR TITLE
Refactor remove line separator to method in the rich-text package.

### PR DESCRIPTION
**Pull request by SergioEstevao**
## Description

This PR refactor the logic for removing a line separator inside a list to use a function on the rich text library.

This allows both the native and the web part to share the same code and simplifies the code on the rich text editor component.

## Types of changes

Refactoring of method.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

File: packages/rich-text/src/remove-line-separator.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ */
+
+import { LINE_SEPARATOR } from './special-characters';
+import { isCollapsed } from './is-collapsed';
+import { remove } from './remove';
+
+/**
+ * Removes a line break character into a Rich Text value at the current
+ * indices. If the Any content between `startIndex` and `endIndex` will be
+ * removed. Indices are retrieved from the selection if none are provided.
```
Comment (etoledom) [inline packages/rich-text/src/remove-line-separator.js]: Might there be a typo in this sentence? `If the Any content between startIndex and endIndex will be removed`

**Review by etoledom — COMMENTED** [review]
Nice step to share more code between mobile and web RichText!

It works well on mobile ✅ But let's wait for @aduth or @ellatrix 's opinion too (or someone else's?), since this also modifies web side files.

A small detail: I think there's a typo in the docs comments. I added it as a code comment.

**Comment by aduth** [comment]
I can review, but @ellatrix would be in the best position to judge whether this change is a sensible addition to the `rich-text` API.

File: packages/rich-text/src/remove-line-separator.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	let removeStart = collapsed ? start - 1 : start;
+	let removeEnd = end;
+	if ( ! backward ) {
+		index = end;
+		removeStart = start;
+		removeEnd = collapsed ? end + 1 : end;
+	}
+
+	if ( text[ index ] !== LINE_SEPARATOR ) {
+		return undefined;
```
Comment (ellatrix) [inline packages/rich-text/src/remove-line-separator.js]: Don't think `undefined` is needed here?

File: packages/rich-text/src/remove-line-separator.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
+/**
+ * Internal dependencies
+ */
+
+import { LINE_SEPARATOR } from './special-characters';
+import { isCollapsed } from './is-collapsed';
+import { remove } from './remove';
+
+/**
+ * Removes a line break character into a Rich Text value at the current
```
Comment (ellatrix) [inline packages/rich-text/src/remove-line-separator.js]: It's not a line break character.

File: packages/rich-text/src/remove-line-separator.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
+/**
+ * Internal dependencies
+ */
+
+import { LINE_SEPARATOR } from './special-characters';
+import { isCollapsed } from './is-collapsed';
+import { remove } from './remove';
+
+/**
```
Comment (ellatrix) [inline packages/rich-text/src/remove-line-separator.js]: Might be good to mention that it only works if there is something to remove.

File: packages/rich-text/src/remove-line-separator.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+ * Removes a line break character into a Rich Text value at the current
+ * indices. If the Any content between `startIndex` and `endIndex` will be
+ * removed. Indices are retrieved from the selection if none are provided.
+ *
+ * @param {Object} value Value to modify.
+ * @param {boolean} backward indicates if are removing from the start index or the end index.
+ *
+ * @return {Object} A new value with the line separator removed. Or undefined if no line separator is found on the position.
```
Comment (ellatrix) [inline packages/rich-text/src/remove-line-separator.js]: The type here does not account for the undefined.

**Review by ellatrix — APPROVED** [review]
Sounds good. Good that it's marked unstable.

**Pull request merged**